js/glowingbear.js

```diff
@@ -656,6 +657,39 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         window.requestAnimationFrame(scroll);
     };
 
+    $scope.parseHost = function() {
+        //The host field is multi purpose for advanced users
+        //There can be a combination of host, port and path
+        //If host is specified here the dedicated port field is disabled
+
+        var parts;
+
+        //host
+        var regexHost = /^([^:\/]*|\[.*\])$/;
+        var regexHostPort = /^([^:]*|\[.*\]):(\d+)$/;
+        var regexHostPortPath = /^([^:]*|\[.*\]):(\d*)\/(.+)$/;
+        if((parts = regexHost.exec(settings.host)) !== null)
+        {
+            settings.hostOnly = parts[1];
+            $rootScope.portDisabled = false;
+        }
+        //host:port
+        else if((parts = regexHostPort.exec(settings.host)) !== null)
+        {
+            settings.hostOnly = parts[1];
+            settings.port = parts[2];
+            $rootScope.portDisabled = true;
+        }
+        //host:port/path
+        else if((parts = regexHostPortPath.exec(settings.host)) !== null)
+        {
+            settings.hostOnly = parts[1];
+            settings.port = parts[2];
+            settings.path = parts[3];
+            $rootScope.portDisabled = true;
+        }
```

Comment:
what if nothing matches and the input is garbage? Maybe we should make the input have a red border to indicate the error or something. I'm sure bootstrap has something for that.

Reply:
Added validation for the input field and also disabled the connect button. Added a red border for all three styles.